Add file filters to the import/export file dialog

yoyokenny opened the pull request:
Adds a file filter to the import/export dialogs so that users can easily tell from a glance that the expected files are in the JSON format.

@Garethp if you'd prefer to set your own file extensions, feel free.

Garethp commented:
Thanks, this is a great idea. I think I'll add .txt too, but I'll get this in a release in the next day or two

yoyokenny commented:
It seems I misunderstood how the extensions were supposed to be entered (the api docs didn't specify)

I've fixed a bug that was caused by that misunderstanding and added .txt to the filters.
